File: engine/engine-model/src/main/scala/tools/packet/foreigneffect/ShowTitleEarned.scala
package tools.packet.foreigneffect

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class ShowTitleEarned(private var _message: String) extends PacketInput {
  def message: String = _message

  override def opcode(): SendOpcode = SendOpcode.SCRIPT_PROGRESS_MESSAGE
}

File: engine/engine-model/src/main/java/tools/packet/foreigneffect/ShowTitleEarned.java
package tools.packet.foreigneffect;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record ShowTitleEarned(String message) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.SCRIPT_PROGRESS_MESSAGE;
   }
}